server/src/test/java/org/elasticsearch/cluster/ClusterStateIT.java

```diff
@@ -1,350 +0,0 @@
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package org.elasticsearch.cluster;
-
-import org.elasticsearch.Version;
-import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse;
-import org.elasticsearch.client.Client;
-import org.elasticsearch.cluster.metadata.IndexGraveyard;
-import org.elasticsearch.cluster.metadata.MetaData;
-import org.elasticsearch.cluster.service.ClusterService;
-import org.elasticsearch.common.CheckedFunction;
-import org.elasticsearch.common.ParseField;
-import org.elasticsearch.common.Priority;
-import org.elasticsearch.common.collect.ImmutableOpenMap;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
-import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Writeable;
-import org.elasticsearch.common.xcontent.NamedXContentRegistry;
-import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.env.Environment;
-import org.elasticsearch.env.NodeEnvironment;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.script.ScriptService;
-import org.elasticsearch.test.ESIntegTestCase;
-import org.elasticsearch.threadpool.ThreadPool;
-import org.elasticsearch.watcher.ResourceWatcherService;
-
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-import java.util.concurrent.atomic.AtomicBoolean;
-
-import static org.elasticsearch.gateway.GatewayService.STATE_NOT_RECOVERED_BLOCK;
-import static org.elasticsearch.test.ESIntegTestCase.Scope.TEST;
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.hamcrest.CoreMatchers.not;
-import static org.hamcrest.Matchers.hasItem;
-import static org.hamcrest.Matchers.instanceOf;
-
-/**
- * This test suite sets up a situation where the cluster has two plugins installed (node, and node-and-transport-client), and a transport
- * client only has node-and-transport-client plugin installed. Each of these plugins inject customs into the cluster state and we want to
- * check that the client can de-serialize a cluster state response based on the fact that the response should not contain customs that the
- * transport client does not understand based on the fact that it only presents the node-and-transport-client-feature.
- */
-@ESIntegTestCase.ClusterScope(scope = TEST)
-public class ClusterStateIT extends ESIntegTestCase {
```

Comment:
@rjernst and I discussed this. This test is testing functionality that is currently only used for the transport client as far as I can see. This is based on the following code:

https://github.com/elastic/elasticsearch/blob/a15f1ee4f64b19670b15a9c1526c7b0b0204ea60/server/src/main/java/org/elasticsearch/cluster/ClusterState.java#L133-L136

At this point in time, the removal of the transport client leaves us with the NodeClient but since the NodeClient is a node, the node must have the same plugins/features that the other nodes have. Given this, I do not see a way for this test to continue to be valid without the transport client. In the future, we may decide to use the feature aware logic for remote cluster connections if we need to serialize the cluster state but this is not in place today which is why I think removal of this test is ok.

@jasontedor is this ok with you?

Reply:
That's fine with me. Thanks for checking.